File: backend/party/party-join/party-join-api/src/main/java/sh/stern/cobralist/party/join/api/JoinPartyService.java
package sh.stern.cobralist.party.join.api;

import sh.stern.cobralist.party.join.domain.FindPartyDTO;
import sh.stern.cobralist.party.join.domain.JoinPartyDTO;

public interface JoinPartyService {
    PartyJoinedDTO joinParty(JoinPartyDTO joinPartyDto);

    FindPartyDTO findParty(String partyCode);
}

File: frontend/src/app/party/join-party/join-party.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {JoinPartyDto} from './join-party.dto';
import {environment} from '../../../environments/environment';
import {Observable} from "rxjs";

interface PartyJoinedDto {
  token: string,
  partyCode: string
}

@Injectable({
  providedIn: 'root'
})
export class JoinPartyService {

  constructor(private httpClient: HttpClient) {
  }

  joinParty(joinPartyDto: JoinPartyDto): Observable<PartyJoinedDto> {
    return this.httpClient.post<PartyJoinedDto>(environment.apiUrl + '/api/party/join/', joinPartyDto);
  }
}
